From: Nikolay Borisov <nik.borisov@suse.com>
Date: Tue, 29 Aug 2023 08:07:25 -0700
Subject: kABI: Reserve extra space for future cpuid/bug ints
Patch-mainline: Never, kabi future proofing
References: git-fixes

To avoid headaches in the future when new cpui leaves or bugs ints are introduced,
lets reserve some extra now, before the kABI is frozen. This will allow for
easier backporting of future patches that require introducing new CPUID/bug ints.


Signed-off-by: Nikolay Borisov <nik.borisov@suse.com>

---
 arch/x86/include/asm/cpufeature.h        |    2 +-
 arch/x86/include/asm/cpufeatures.h       |    4 ++--
 tools/arch/x86/include/asm/cpufeatures.h |    4 ++--
 3 files changed, 5 insertions(+), 5 deletions(-)

--- a/arch/x86/include/asm/cpufeature.h
+++ b/arch/x86/include/asm/cpufeature.h
@@ -35,7 +35,7 @@ enum cpuid_leafs
 	CPUID_8000_001F_EAX,
 	CPUID_8000_0021_EAX,
 	CPUID_LNX_5,
-	NR_CPUID_WORDS,
+	NR_CPUID_WORDS = 25,
 };
 
 #define X86_CAP_FMT_NUM "%d:%d"
--- a/arch/x86/include/asm/cpufeatures.h
+++ b/arch/x86/include/asm/cpufeatures.h
@@ -5,8 +5,8 @@
 /*
  * Defines x86 CPU feature bits
  */
-#define NCAPINTS			22	   /* N 32-bit words worth of info */
-#define NBUGINTS			2	   /* N 32-bit bug flags */
+#define NCAPINTS			25	   /* N 32-bit words worth of info */
+#define NBUGINTS			4	   /* N 32-bit bug flags */
 
 /*
  * Note: If the comment begins with a quoted string, that string is used
--- a/tools/arch/x86/include/asm/cpufeatures.h
+++ b/tools/arch/x86/include/asm/cpufeatures.h
@@ -5,8 +5,8 @@
 /*
  * Defines x86 CPU feature bits
  */
-#define NCAPINTS			22	   /* N 32-bit words worth of info */
-#define NBUGINTS			2	   /* N 32-bit bug flags */
+#define NCAPINTS			25	   /* N 32-bit words worth of info */
+#define NBUGINTS			4	   /* N 32-bit bug flags */
 
 /*
  * Note: If the comment begins with a quoted string, that string is used
